test/session/SessionSpec.coffee

```diff
@@ -772,3 +772,45 @@ describe 'Session', ->
             resolve()))
 
     .then((() -> done()), (err) -> done(err))
+
+  it 'should limit the number of sessions', (done) ->
+    [alice_ident, bob_ident] = [0, 1].map(-> Proteus.keys.IdentityKeyPair.new())
+    bob_store = new TestStore Proteus.keys.PreKey.generate_prekeys 0, (Proteus.session.Session.MAX_SESSION_STATES + 2)
+
+    bob_bundle = (i, store) -> Proteus.keys.PreKeyBundle.new bob_ident.public_key, store.prekeys[i]
+
+    alice = null
+    bob = null
+    hello_bob = null
+
+    Proteus.session.Session.init_from_prekey alice_ident, bob_bundle 1, bob_store
+    .then (s) ->
+      alice = s
+      alice.encrypt 'Hello Bob!'
+
+    .then (m) ->
```

Comment:
Please use proper naming (like `message`) instead of cryptic letters (like `m`).